akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1021,12 +1021,12 @@ trait FlowOps[+Out, +Mat] {
   def splitWhen(p: Out ⇒ Boolean): SubFlow[Out, Mat, Repr, Closed] = {
     val merge = new SubFlowImpl.MergeBack[Out, Repr] {
       override def apply[T](flow: Flow[Out, T, Unit], breadth: Int): Repr[T] =
-        deprecatedAndThen[Source[Out, Unit]](Split.when(p.asInstanceOf[Any ⇒ Boolean]))
-          .map(_.via(flow))
+        via(Split.when(p))
+          .map(_.buffer(16, OverflowStrategy.backpressure).via(flow))
```

Comment:
this buffer is in there only for debug purposes, should not be in the final solution.